### PR TITLE
refactor: use effect Micro for the client

### PR DESCRIPTION
**Pull request by tim-smart**
- **add flake**
- **refactor: try effect Micro**
- **strip effect from bundle**

### CODE REVIEW

File: .github/analyze-bundle.js [hunk 1]
```diff
@@ -53,68 +53,79 @@ async function getTotalBundleSize(filepath) {
...
+  console.log(`mainGzip: ${mainGzip}`);
+  console.log(`prGzip: ${prGzip}`);
+  console.log(`diff: ${formatDiff(prGzip - mainGzip)}`);
+
+  // Upload HTML files for easy inspection (not on forks)
+  let files = [];
+  if (
+    typeof process.env.UPLOADTHING_SECRET === "string" &&
+    process.env.UPLOADTHING_SECRET.length > 0
+  ) {
```
Comment (juliusmarminge) [inline .github/analyze-bundle.js:77]: why can't we have nice things... any way to make this work for forks too cc @markflorkowski

File: packages/shared/src/effect.ts [hunk 2]
```diff
@@ -16,11 +14,10 @@ export type FetchContextService = {
     "x-uploadthing-be-adapter": string | undefined;
   };
 };
-export class FetchContext
-  extends /** #__PURE__ */ Context.Tag("uploadthing/FetchContext")<
-    FetchContext,
-    FetchContextService
-  >() {}
+export class FetchContext extends Context.Tag("uploadthing/FetchContext")<
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply packages/shared/src/effect.ts:17]: Are there some cases where this isn't used? Are you setting `sideEffects` in the package.json?

File: packages/uploadthing/src/internal/handler.ts [hunk 3]
```diff
@@ -108,12 +108,14 @@ export const buildRequestHandler =
...
       Effect.catchTags({
-        InvalidJsonError: (e) =>
-          new UploadThingError({
-            code: "INTERNAL_SERVER_ERROR",
-            message: "An error occured while parsing input/output",
-            cause: e,
-          }),
+        InvalidJson: (e) =>
+          Effect.fail(
+            new UploadThingError({
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply packages/uploadthing/src/internal/handler.ts:113]: They are yieldable again, but there was one point in the PR where Micro.TaggedError didn't exist.

File: examples/backend-adapters/server/src/effect.ts [hunk 4]
```diff
@@ -11,36 +21,36 @@ const { GET, POST } = createRouteHandler({
...
-  HttpServer.router.get("/api", HttpServer.response.text("Hello from Effect")),
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/api", HttpServerResponse.text("Hello from Effect")),
 
-  HttpServer.router.get(
+  HttpRouter.get(
     "/api/uploadthing",
-    HttpServer.request.ServerRequest.pipe(
-      Effect.map(BunHttpServer.request.toRequest),
+    HttpServerRequest.HttpServerRequest.pipe(
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply examples/backend-adapters/server/src/effect.ts]: The server apis have changed recently.

File: packages/shared/src/effect.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/shared/src/effect.ts:17]: We use this package on our infra side where some packages might only import a utility like `isObject`

File: packages/uploadthing/src/internal/handler.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/uploadthing/src/internal/handler.ts:113]: cool i'll swap them back then 👍

**Review by juliusmarminge — APPROVED** [review]
👏

**Pull request merged**